[CI] APMYamlTestSuiteIT test {yaml=/20_metrics_ingest/Test metrics-apm.app-* setting event.ingested via ingest pipeline} failing

**Build Scans:**
- [elasticsearch-intake #14623 / part2](https://gradle-enterprise.elastic.co/s/uebdnewu6aosm)
- [elasticsearch-periodic-platform-support #5374 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/3323ys4lftvkk)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:apm-data:yamlRestTest" --tests "org.elasticsearch.xpack.apmdata.APMYamlTestSuiteIT" -Dtests.method="test {yaml=/20_metrics_ingest/Test metrics-apm.app-* setting event.ingested via ingest pipeline}" -Dtests.seed=F94B1CDFB7B92159 -Dtests.locale=sr-Latn-ME -Dtests.timezone=Etc/GMT+5 -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.apmdata.APMYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Byaml%3D%2F20_metrics_ingest%2FTest%20metrics-apm.app-*%20setting%20event.ingested%20via%20ingest%20pipeline%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [/20_metrics_ingest:32]: field [hits.hits.1.fields.event\.ingested.0] is null
```

**Issue Reasons:**
- [8.x] 2 failures in test test {yaml=/20_metrics_ingest/Test metrics-apm.app-* setting event.ingested via ingest pipeline} (0.6% fail rate in 356 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

> [@inge4pres](https://github.com/inge4pres) I believe this is related to the recent APM work. Can you take a look at it?

Any hints on why ingest pipeline would behave differently than before?

We introduced a change in how we populate APM data in 8.16 and 8.17 with https://github.com/elastic/elasticsearch/pull/119209 and
https://github.com/elastic/elasticsearch/pull/119208 and we haven't seen such failures happen.

The same change is also in `main`, so should we expect failures also for 8.18?

The failure message is suspicious

```
java.lang.AssertionError: Failure at [/20_metrics_ingest:32]: field [hits.hits.1.fields.event\.ingested.0] is null
```

I don't see how it could be null, since we populate it with the ingest pipeline

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -217,8 +217,6 @@ tests:
 - class: org.elasticsearch.discovery.ec2.DiscoveryEc2AvailabilityZoneAttributeNoImdsIT
   method: testAvailabilityZoneAttribute
   issue: https://github.com/elastic/elasticsearch/issues/118564
-- class: org.elasticsearch.xpack.apmdata.APMYamlTestSuiteIT
-  method: test {yaml=/20_metrics_ingest/Test metrics-apm.app-* setting event.ingested via ingest pipeline}
   issue: https://github.com/elastic/elasticsearch/issues/118875
 - class: org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT
   method: testSeqNoCASLinearizability

